tests/SpecTests/TransactionsSpecTest.php

```diff
@@ -151,7 +155,7 @@ public function testTransactions(stdClass $test, array $runOn = null, array $dat
         $databaseName = isset($databaseName) ? $databaseName : $this->getDatabaseName();
         $collectionName = isset($collectionName) ? $collectionName : $this->getCollectionName();
 
-        $context = Context::fromTransactions($test, $databaseName, $collectionName);
+        $context = Context::fromTransactions($test, $databaseName, $collectionName, $useMultipleMongoses);
```

Comment:
This looks fine. Alternatively, we could inject the `$uri` instead and insulate `Context::fromTransactions()` (and `fromRetryableWrites()`) from having to think about `$useMultipleMongoses`. Up to you.

Reply:
I thought about this as well, but ultimately decided on this solution. Since `useMultipleMongoses` is a context option for the tests, it makes most sense to handle this internally and not leak it into the tests themselves. Since the context creates its own client and currently knows where to get the URL from, I thought it was best to leave that logic encapsulated and simply expose the `useMultipleMongoses` argument in the factories.